nucliadb/src/nucliadb/search/search/pgcatalog.py

```diff
@@ -166,10 +168,18 @@ async def pgcatalog_search(query_parser: QueryParser) -> Resources:
                 tmp_facets: dict[str, dict[str, int]] = {
                     translate_label(f): {} for f in query_parser.faceted
                 }
+                facet_filters = " OR ".join(f"label LIKE '{f}/%%'" for f in tmp_facets.keys())
+                for facet in tmp_facets.keys():
+                    if not (
+                        facet.startswith("/n/s") or facet.startswith("/n/i") or facet.startswith("/l")
+                    ):
+                        logger.warn(f"Unexpected facet used at catalog: {facet}")
+
                 await cur.execute(
-                    f"SELECT unnest(labels) AS label, COUNT(*) FROM ({query}) fc GROUP BY 1 ORDER BY 1",
+                    f"SELECT label, COUNT(*) FROM (SELECT unnest(labels) AS label FROM ({query}) fc) nl WHERE ({facet_filters}) GROUP BY 1 ORDER BY 1",
```

Comment:
should we order by 2?

Reply:
We order by 1 (alphabetical order) to keep previous behaviour. It also makes the python post-filtering easier.